features/filesystem/bd/FlashSimBlockDevice.cpp

```diff
@@ -0,0 +1,142 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "FlashSimBlockDevice.h"
+#include "mbed_assert.h"
+#include <algorithm>
+#include <stdlib.h>
+
+static const uint8_t blank = 0xFF;
+static const bd_size_t min_blank_buf_size = 32;
```

Comment:
I think even `min_blank_buf_size` should be configurable

Reply:
Can you explain? Was trying to define a value that should be small enough, so it wouldn't effect heap usage. Do you think it should be smaller? A larger value can make the blank programming/checking a bit more efficient, but this seems negligible to me.

Reply:
It depends on the use case of this class. If you have it as variable, we keep it configurable for application to have more flexibility.  

Ideally for me it should be equivalent to program size and not separate parameter. (As in case of real flash only if entire programmable block is empty than we can say it is free and not by checking few bytes of block)

Reply:
Yeah I am kind of confused with the usage of this buffer, instead of having it equivalent to program size it is less to save on heap space, and since it is RAM the iterations will be fast. But in the test example program size (8) is less then blank buffer size (32), shouldn't we cap it to program size.
If you would have used defines, and static allocation I would not have made this argument, but we are using dynamic allocation here.

Reply:
Capping it to program size will make it extremely inefficient if program size is 1 (saw flash devices with such parameters), meaning that both program and erase functions will read/write chunks of 1. Hence the minimum of 32, which should work even if program size is 1.